From 803c69b2f7b476f904db970db8d167025c2d9845 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 22 Aug 2017 16:56:24 +0800
Subject: [PATCH 2436/5242] MLK-16242-7 video: fbdev: dcss: add non-blocking
 mode in pan display

commit  01a8d75bda1c5636b5e5e551425b7cd5245db790 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add non-blocking mode for pan display interface to
support very special cases to focus on the maximum
FPS display.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 7f6ce99..89898c2 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2550,8 +2550,10 @@ static int dcss_pan_display(struct fb_var_screeninfo *var,
 		return ret;
 	}
 
-	/* make pan display synchronously */
-	flush_workqueue(info->ctxld_wq);
+	/* TODO: blocking mode */
+	if (likely(!var->reserved[2]))
+		/* make pan display synchronously */
+		flush_workqueue(info->ctxld_wq);
 
 	return 0;
 }
-- 
1.7.9.5

